docs/the-guide/chapter-7-track-model-evolutions-in-the-cicd-pipeline-with-cml/index.md

```diff
@@ -478,10 +486,13 @@ merge requests (MRs) - to integrate the work done into the `main` branch.
 
 	 variables:
 	   # Change pip's cache directory to be inside the project directory since we can
-	@@ -33,3 +34,72 @@ train:
+	@@ -38,4 +39,73 @@ train:
+	     # Pull data from DVC
 	     - dvc pull
 	     # Run the experiment
-	     - dvc repro
+	-    - dvc repro
+	\ No newline at end of file
```

Comment:
Have we missed this line in the original file?

Reply:
This is normal as the original GitHub ci file ends with a newline, and in the update one at chapter 7, we add the new content after it removing the newline.

Reply:
I think you can add it, it's the only time in the guide that a git diff shows a missing new line so we can add it manually if needed so the message `\ No newline at end of file` doesn't show up in here.

Reply:
But then the pipeline would have a weird structure:

```yaml
...

jobs:
  train:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3
      - name: Install poetry
        run: pipx install poetry==1.4.0
      - name: Setup Python
        uses: actions/setup-python@v4
        with:
          python-version: '3.10'
          cache: 'poetry'
      - name: Install dependencies
        run: poetry install
      - name: Setup DVC
        uses: iterative/setup-dvc@v1
        with:
          version: '2.37.0'
      - name: Login to Google Cloud
        uses: 'google-github-actions/auth@v1'
        with:
          credentials_json: '${{ secrets.GCP_SERVICE_ACCOUNT_KEY }}'
      - name: Train model
        run: |
          # Pull data from DVC
          dvc pull
          # Run the experiment
          dvc repro

      - name: Upload evaluation results
        uses: actions/upload-artifact@v3
        with:
          path: evaluation
          retention-days: 5
```

Reply:
Check out my other comment, if you add the new line there, it will resolve the `\ No newline at end of file` here and make a more natural Git diff. :)